sklearn/linear_model/ridge.py

```diff
@@ -141,7 +145,16 @@ def _solve_cholesky_kernel(K, y, alpha, sample_weight=None):
         # Only one penalty, we can solve multi-target problems in one time.
         K.flat[::n_samples + 1] += alpha[0]
 
-        dual_coef = linalg.solve(K, y, sym_pos=True, overwrite_a=True)
+        try:
+            # Note: we must use overwrite_a=False in order to be able to
+            #       use the fall-back solution below in case a LinAlgError
+            #       is raised
+            dual_coef = linalg.solve(K, y, sym_pos=True,
+                                     overwrite_a=False)
+        except np.linalg.LinAlgError:
+            warnings.warn("Singular matrix in solving dual problem. Using "
+                          "least-squares solution instead.")
+            dual_coef = linalg.lstsq(K, y)[0]
```

Comment:
@jmetzen if and only if it is easy to add to your bench: For completeness, the runtime of the `pinv2` option is interesting. Who knows, there may be a surprise :) 

Thanks also for checking back in scipy release history for the multitarget functionality of `lstsq`. It is true that I only checked on my computer with 0.14 before I proposed it.